File: ts/src/test/base/test.parsePrecision.ts

// AUTO_TRANSPILE_ENABLED

import assert from 'assert';
import ccxt from '../../../ccxt.js';

function testParsePrecision () {

    const exchange = new ccxt.Exchange ({
        'id': 'sampleexchange',
    });

    assert (exchange.parsePrecision ('15') === '0.000000000000001');
    assert (exchange.parsePrecision ('1') === '0.1');
    assert (exchange.parsePrecision ('0') === '1');
    assert (exchange.parsePrecision ('-5') === '100000');
}

export default testParsePrecision;

File: julia/Ccxt/test/base/test.parsePrecision.jl
using Test
using Ccxt
function testParsePrecision()

    exchange = get(ccxt, Symbol("Exchange"), nothing)(Dict{Symbol, Any}(
        Symbol("id") => "sampleexchange"
    ));
    @test parsePrecision(exchange, "15") == "0.000000000000001";
    @test parsePrecision(exchange, "1") == "0.1";
    @test parsePrecision(exchange, "0") == "1";
    @test parsePrecision(exchange, "-5") == "100000";
end
